<?xml version="1.0"?>
<!-- Provided by sleutene for convenience -->
<launch>
  <!-- IPv4 address of your drone -->
  <arg name="ip" default="192.168.1.1" />
  <!-- Ultrasound frequency (7 or 8). -->
  <arg name="freq" default="8" />
  <!-- Launch ardrone_driver. -->
  <node name="ardrone_driver" pkg="ardrone_autonomy" type="ardrone_driver"
      output="screen" clear_params="true" args="-ip $(arg ip)">
    <param name="outdoor" value="0" />
    <param name="max_bitrate" value="4000" />
    <param name="bitrate" value="4000" />
    <param name="navdata_demo" value="0" />
    <param name="flight_without_shell" value="0" />
    <param name="altitude_max" value="1600" />
    <param name="altitude_min" value="50" />
    <param name="euler_angle_max" value="0.21" />
    <param name="control_vz_max" value="700" />
    <param name="control_yaw" value="1.75" />
    <param name="detect_type" value="10" />
    <param name="enemy_colors" value="3" />
    <param name="detections_select_h" value="32" />
    <param name="detections_select_v_hsync" value="128" />
    <param name="enemy_without_shell" value="0" />
    <param name="ultrasound_freq" value="$(arg freq)" />
    <param name="realtime_navdata" value="true" />
    <param name="realtime_video" value="true" />
    <!-- Covariance Values (3x3 matrices reshaped to 1x9)-->
    <rosparam param="cov/imu_la">[0.1, 0.0, 0.0, 0.0, 0.1, 0.0, 0.0, 0.0, 0.1]</rosparam>
    <rosparam param="cov/imu_av">[1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0]</rosparam>
    <rosparam param="cov/imu_or">[1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 100000.0]</rosparam>
  </node>
  <!-- Launch visualisation. -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ardrone_practicals)/config/rviz_vi_ekf.rviz"/>

    <!-- Set Log Level, possible levels: debug, info, warn, error, fatal -->
  <node pkg="rosservice" type="rosservice" name="set_node_log_level" 
      args="call /arp_node/set_logger_level '{logger: 'ros.ardrone_practicals', level: 'info'}'" />
  <node pkg="rosservice" type="rosservice" name="set_log_level" 
      args="call /arp_node/set_logger_level '{logger: 'ros.ardrone_practicals.custom', level: 'info'}'" />

  <!-- Launch estimator / control arp_node. -->
  <node name="arp_node" pkg="ardrone_practicals" type="arp_node"
      output="screen">
		<!-- remap from="/cmd_vel" to="/cmd_vel_test"/>-->

    <!-- Global Variables -->
    <param name="enableFusion" type="bool" value="true" />
    <param name="camModel" type="bool" value="false" />
    <param name="displayKeypoints" type="bool" value="true" />
    <param name="displayAllKeypoints" type="bool" value="false" />
    <param name="numKeypoints" type="int" value="200" />
    <param name="poseLostThreshold" value="100" />
    <param name="poseSwitchThreshold" value="5" />
    <param name="poseTimeThreshold" value="2" />
    <param name="Brisk_uniformityRadius" value="35" />
    <param name="Brisk_absoluteThreshold" value="20" />
    <param name="ImageWidth" value="1920" />
    <param name="ImageHeight" value="960" />
    <param name="fontScaling" value="1.5" />
    
    <!-- Camera parameters-->
    <param name="fu" value="582.047949" />
    <param name="fv" value="579.300111" />
    <param name="cu" value="316.339374" />
    <param name="cv" value="205.835304" />
    <param name="k1" value="-0.562099" />
    <param name="k2" value="0.358005" />
    <param name="p1" value="-0.011755" />
    <param name="p2" value="-0.001345" />
    <rosparam param="T_SC">[-0.00195087, -0.03257782,  0.99946730, 0.17409445, -0.99962338, -0.02729525, -0.00284087, 0.02255834, 0.02737326, -0.99909642, -0.03251230, 0.00174723, 0.00000000,  0.00000000,  0.00000000, 1.00000000]</rosparam>
    <!-- Make sure to load the right map! -->
    <param name="map" value="okvis2-slam-final_map_dronelab.csv" />
    <param name="occupancymap" value="occupancy-map_garching.dat" />

  </node>

</launch>
